Loop/Managers/LoopDataManager.swift

```diff
@@ -1595,6 +1602,20 @@ extension LoopDataManager {
             dispatchPrecondition(condition: .onQueue(loopDataManager.dataAccessQueue))
             return try loopDataManager.recommendBolusForManualGlucose(glucose, consideringPotentialCarbEntry: potentialCarbEntry, replacingCarbEntry: replacedCarbEntry)
         }
+        
+        func didInsulinCounteractionEffectsReachThreshold(_ threshold: HKQuantity, startDate: Date, endDate: Date) -> Bool {
```

Comment:
[nit] I think it would be best if the concerns around this first-stab "meal detector" were kept separate from LoopDataManager. Its input is recent ICE, and its output is binary flag. And all the details, like how much ice, the threshold, how the averaging is done, etc, are internal.

Reply:
If I'm understanding you correctly, I'll use LoopDataManager to just get the ICE and conduct that calculation in the carb entry VC.

Reply:
Yes, or a separate model class; something like "SimpleMealDetector"